test/e2e/petset.go

```diff
@@ -177,6 +209,140 @@ var _ = framework.KubeDescribe("PetSet [Slow] [Feature:PetSet]", func() {
 		})
 	})
 
+	framework.KubeDescribe("Scale up/down testing [Slow] [Feature:PetSet]", func() {
+		psName := "pet"
+		labels := map[string]string{
+			"name": psName,
+		}
+		headlessSvcName := "test"
+		var controller *cache.Controller
+		var stopCh chan struct{}
+		var podTracker []podEvent
+
+		BeforeEach(func() {
+			framework.SkipUnlessProviderIs("gce", "vagrant")
+
+			By("creating service " + headlessSvcName + " in namespace " + ns)
+			headlessService := createServiceSpec(headlessSvcName, "", true, labels)
+			_, err := c.Services(ns).Create(headlessService)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("configuring controller to watch for added pods")
+			labelSelector, err := unversioned.LabelSelectorAsSelector(&unversioned.LabelSelector{
+				MatchLabels: labels,
+			})
+			ExpectNoError(err)
```

Comment:
nit: which function is this (I think there's a similar function in framework.ExpectNoError)? is there a reason it needs to be global to the e2e package if it's a petset specific thing ? (sorry this is probably my fault but I just noticed it in this pr)